typescript/infra/src/core/govern.ts

```diff
@@ -42,39 +49,91 @@ export class AbacusCoreGovernor<Chain extends ChainName> {
+      objMap(this.calls, async (chain, calls) => {
+        const connection = connectionFn(chain);
+        await executeFn(chain, connection, calls);
+      }),
+    );
+  }
+
+  protected connectionFn(chain: Chain) {
+    return this.checker.multiProvider.getChainConnection(chain);
+  }
+
+  protected ledgerConnectionFn(chain: Chain) {
+    const connection = this.checker.multiProvider.getChainConnection(chain);
+    return new ChainConnection({
+      signer: new LedgerSigner(connection.provider),
+      provider: connection.provider,
+      overrides: connection.overrides,
+      confirmations: connection.confirmations,
     });
   }
 
+  protected async estimateFn(
+    chain: Chain,
+    connection: ChainConnection,
+    calls: types.CallData[],
+  ) {
+    const signer = connection.signer;
+    if (!signer) throw new Error(`no signer found for ${chain}`);
+    const from = await signer.getAddress();
+    await Promise.all(
+      calls.map((call) =>
+        signer.estimateGas({
+          ...call,
+          from,
+        }),
+      ),
+    );
+  }
+
+  protected async sendFn(
+    chain: Chain,
+    connection: ChainConnection,
+    calls: types.CallData[],
+  ) {
+    const signer = connection.signer;
+    if (!signer) throw new Error(`no signer found for ${chain}`);
+    for (const call of calls) {
+      const response = await signer.sendTransaction(call);
+      console.log(`sent tx ${response.hash} to ${chain}`);
+      await response.wait(connection.confirmations);
+      console.log(`confirmed tx ${response.hash} on ${chain}`);
+    }
+  }
+
   estimateCalls() {
-    objMap(this.calls, async (chain, calls) => {
-      const connection = this.checker.multiProvider.getChainConnection(chain);
-      const signer = connection.signer;
-      if (!signer) {
-        throw new Error(`signer not found for ${chain}`);
-      }
-      for (const call of calls) {
-        await signer.estimateGas({ ...call, from: await signer.getAddress() });
-      }
-    });
+    return this.executeCalls(this.connectionFn, this.estimateFn);
   }
 
-  async executeCalls() {
-    await this.estimateCalls();
-    objMap(this.calls, async (chain, calls) => {
-      const connection = this.checker.multiProvider.getChainConnection(chain);
-      const signer = connection.signer;
-      if (!signer) {
-        throw new Error(`signer not found for ${chain}`);
-      }
-      for (const call of calls) {
-        const response = await signer.sendTransaction(call);
-        console.log(`sent tx ${response.hash} to ${chain}`);
-        await response.wait(connection.confirmations);
-        console.log(`confirmed tx ${response.hash} on ${chain}`);
-      }
-    });
+  sendCalls() {
+    return this.executeCalls(this.connectionFn, this.sendFn);
+  }
+
+  estimateCallsLedger() {
+    return this.executeCalls(this.ledgerConnectionFn, this.estimateFn);
+  }
+
+  sendCallsLedger() {
+    return this.executeCalls(this.ledgerConnectionFn, this.sendFn);
```

Comment:
if this is all on `ChainConnection` we don't need these

Reply:
Why is that?